src/app/sto/components/ConfigureSTO.js

```diff
@@ -11,61 +11,110 @@ import {
   ModalHeader,
   Icon,
 } from 'carbon-components-react'
-import type { SecurityToken, STOFactory } from 'polymathjs/types'
+import BigNumber from 'bignumber.js'
+import { Remark } from 'polymath-ui'
+import type { SecurityToken, STOFactory, Address } from 'polymathjs/types'
 
 import NotFoundPage from '../../NotFoundPage'
 import STODetails from './STODetails'
 import ConfigureSTOForm from './ConfigureSTOForm'
-import { configure, goBack } from '../actions'
+import { configure, goBack, faucet, getPolyFee } from '../actions'
 import type { RootState } from '../../../redux/reducer'
 
 type StateProps = {|
+  account: ?string,
   token: ?SecurityToken,
-  factory: ?STOFactory
+  factory: ?STOFactory,
+  networkName: string,
+  polyBalance: BigNumber
 |}
 
 type DispatchProps = {|
-  configure: () => any,
-  goBack: () => any
+  configure: (number, Address) => any,
+  goBack: () => any,
+  faucet: (? string, number) => any,
+  getPolyFee: () => any
 |}
 
 const mapStateToProps = (state: RootState): StateProps => ({
+  account: state.network.account,
   token: state.token.token,
   factory: state.sto.factory,
+  networkName: state.network.name,
+  polyBalance: state.pui.account.balance,
 })
 
 const mapDispatchToProps: DispatchProps = {
   configure,
   goBack,
+  faucet,
+  getPolyFee,
 }
 
 type Props = {||} & StateProps & DispatchProps
 
 type State = {|
-  isModalOpen: boolean,
+  isConfirmationModalOpen: boolean,
+  isNotEnoughPolyModalOpen: boolean,
+  polyCost: number,
+  fundsReceiver: Address,
+  isConfirmationModal2Open: boolean,
 |}
 
 class ConfigureSTO extends Component<Props, State> {
 
   state = {
-    isModalOpen: false,
+    isConfirmationModalOpen: false,
+    isNotEnoughPolyModalOpen: false,
+    polyCost: 0,
+    fundsReceiver: '',
+    isConfirmationModal2Open: false,
   }
 
-  handleSubmit = () => {
-    this.setState({ isModalOpen: true })
+  componentWillMount () {
+    this.props.getPolyFee().then((fee) => {
+      this.setState({ polyCost: Number(fee) })
```

Comment:
you should do this in action creators and store such values in redux state, not react state